src/natcap/invest/seasonal_water_yield/seasonal_water_yield_core.pyx

```diff
@@ -621,6 +621,14 @@ cpdef calculate_local_recharge(
                         qf_nodata = qf_m_nodata_list[m_index]
                         kc_nodata = kc_m_nodata_list[m_index]
 
+                        # make sure that user input nodata values are defined
+                        # precipitation and evapotranspiration data should 
+                        # always be non-negative
+                        if precip_nodata is None:
+                            precip_nodata = -1
+                        if et0_nodata is None:
+                            et0_nodata = -1
+
```

Comment:
I was thinking about this over the weekend.  Checking for and assigning values for  `precip_nodata` and `et0_nodata` here will make these extra lines run many, many times (once for each month, but also for each pixel in the `work_queue`, which ends up being for each valid pixel).  Of course it's quite possible that the compiler will optimize this condition away so that it doesn't need to be run so many times, but since we know in advance that this can just be defined elsewhere, would you be ok with moving it outside of the main processing loop?  Offhand, it looks like these could be checked for once, up near the top of this function (around line 448 for `et0`, 456 for `precip`).

Reply:
That's a good point! Just moved it outside of the loop